lesson_26/api/java/api_app/src/main/java/com/codedifferently/lesson26/web/MediaItemsController.java

```diff
@@ -29,4 +34,31 @@ public GetMediaItemsResponse getItems() {
     var response = GetMediaItemsResponse.builder().items(responseItems).build();
     return response;
   }
+
+  @PostMapping("/items")
+  public MediaItemResponse addItemById(@RequestBody CreateMediaItemRequest createRequest) {
+    if (createRequest.getItem().getId() == null) {
+      createRequest.getItem().setId(UUID.randomUUID());
+    }
+    
+      library.add(createRequest.getItem());
+    
+    return  MediaItemResponse.from(createRequest.getItem());
+  }
+
+  @GetMapping("/items/:{id}")
+  public MediaItemResponse getItemById(@PathVariable("id") UUID id) {
+    MediaItem item = library.search(SearchCriteria.builder().id(id.toString()).build()).stream().findFirst().orElseThrow();
+   
+    return MediaItemResponse.from(item);
+  }
+
+  @DeleteMapping("/items/:{id}") 
+  public MediaItemResponse deleteItemById(@PathVariable("id") UUID id) {
```

Comment:
Simila to the above note re: `ResponseEntity`, you can use that to return a `204 No Content` or `404 Not Found` response.